src/utils/self_consistent_field.jl

```diff
@@ -14,17 +14,16 @@ function self_consistent_field(ham::Hamiltonian, n_bands::Int, n_electrons::Int;
                                ρ=nothing, tol=1e-6, T=0, smearing=nothing,
                                lobpcg_prec=PreconditionerKinetic(ham, α=0.1),
                                max_iter=100, algorithm=:scf_nlsolve, kwargs...)
-    if smearing === nothing
-        compute_occupation(basis, energies, Psi) = occupation_step(basis, energies, Psi,
-                                                                   n_electrons)
+    if smearing == nothing
```

Comment:
Tests for nothing should have `===`, see e.g. https://docs.julialang.org/en/latest/base/base/#Base.isnothing

Reply:
Hm, you're right. Should not matter in principle but the compiler is sure that `x === nothing` is false as long as the type of `x` is not Nothing, whereas `==` might do stranger things. Thanks!